src/naive/datetime/mod.rs

```diff
@@ -127,6 +132,40 @@ impl NaiveDateTime {
         datetime.expect("invalid or out-of-range datetime")
     }
 
+    /// Creates a new [NaiveDateTime] from milliseconds since the UNIX epoch.
+    ///
+    /// The UNIX epoch starts on midnight, January 1, 1970, UTC.
+    ///
+    /// Returns `None` on an out-of-range number of milliseconds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chrono::NaiveDateTime;
+    /// let timestamp_millis: i64 = 1662921288; //Sunday, September 11, 2022 6:34:48 PM
+    /// let naive_datetime = NaiveDateTime::from_timestamp_millis(timestamp_millis);
+    /// assert!(naive_datetime.is_some());
+    /// assert_eq!(timestamp_millis, naive_datetime.unwrap().timestamp_millis());
+    ///
+    /// // Negative timestamps (before the UNIX epoch) are supported as well.
+    /// let timestamp_millis: i64 = -2208936075; //Mon Jan 01 1900 14:38:45 GMT+0000
+    /// let naive_datetime = NaiveDateTime::from_timestamp_millis(timestamp_millis);
+    /// assert!(naive_datetime.is_some());
+    /// assert_eq!(timestamp_millis, naive_datetime.unwrap().timestamp_millis());
+    /// ```
+    #[inline]
+    pub fn from_timestamp_millis(millis: i64) -> Option<NaiveDateTime> {
+        let secs = millis / 1000;
+
+        if millis < 0 {
+            let nsecs = (millis % 1000).abs() as u32 * NANOS_IN_MILLISECOND;
+            NaiveDateTime::from_timestamp_opt(secs - 1, NANOS_IN_SECOND - nsecs)
```

Comment:
Thanks for the impl that handles negative millis. If possible could you subtract with [`checked_sub`](https://doc.rust-lang.org/std/primitive.i64.html#method.checked_sub) here to avoid potential overflow

Reply:
Done